driver-sync/src/main/com/mongodb/client/ClientSession.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client;
+
+import com.mongodb.TransactionOptions;
+
+/**
+ * A client session that supports transactions.
+ *
+ * @since 3.8
+ */
+public interface ClientSession extends com.mongodb.session.ClientSession {
```

Comment:
Adding this new interface was the toughest decision to make, as it breaks both source and binary compatibility:

* MongoClient.startSession now returns an instance of this instead of com.mongodb.session.ClientSession, which breaks binary but not source compatibility
* MongoCollection overloads now take an instance of this instead of com.mongodb.session.ClientSession, which breaks both source and binary compatibility.